src/core/service.c

```diff
@@ -536,24 +536,9 @@ static int service_verify(Service *s) {
         if (UNIT(s)->load_state != UNIT_LOADED)
                 return 0;
 
-        for (ServiceExecCommand c = 0; c < _SERVICE_EXEC_COMMAND_MAX; c++) {
-                ExecCommand *command;
-
-                LIST_FOREACH(command, command, s->exec_command[c]) {
-                        if (!path_is_absolute(command->path) && !filename_is_valid(command->path)) {
-                                log_unit_error(UNIT(s),
-                                               "Service %s= binary path \"%s\" is neither a valid executable name nor an absolute path. Refusing.",
-                                               command->path,
-                                               service_exec_command_to_string(c));
-                                return -ENOEXEC;
-                        }
-                        if (strv_isempty(command->argv)) {
-                                log_unit_error(UNIT(s),
-                                               "Service has an empty argv in %s=. Refusing.",
-                                               service_exec_command_to_string(c));
-                                return -ENOEXEC;
-                        }
-                }
+        if (!service_unit_name_is_valid(UNIT(s)->id)) {
```

Comment:
This hunk is wrong: it should only add a new check.

Reply:
Hunk remain in place, while that new check was removed in 3e1db806b0c18fd6138886ce67fac2655f09caef.